src/main/java/pro/java/hw17/FileLogger.java

```diff
@@ -0,0 +1,153 @@
+            String timestamp = dateFormat.format(new Date());
+            String logMessage = String.format("[%s][%s] %s: %s", timestamp, level, "Message", message);
+            writer.println(logMessage);
+            writer.flush();
+            currentFileSize += logMessage.length();
+        } else {
+            System.out.println("Log message is not logged due to wrong logging level: " + level);
+        }
+    }
+
+    public void warning(String message) {
+        try {
+            if (config.isLogToConsole()) {
+                String logMessage = "[" + getTimestamp() + "][" + LoggingLevel.WARN + "] " + "Message: " + message;
+                System.out.println(logMessage);
+            } else {
+                log(message, LoggingLevel.WARN);
+            }
+        } catch (FileMaxSizeReachedException e) {
+            displayFileMaxSizeMessages(e);
+        } catch (IOException e) {
+            System.err.println("Error occurred while logging WARNING message: " + message);
+        }
+    }
+
+    public void info(String message) {
+        if (LoggingLevel.INFO.ordinal() <= config.getLogLevel().ordinal()) {
+            try {
+                if (config.isLogToConsole()) {
+                    String logMessage = "[" + getTimestamp() + "][" + LoggingLevel.INFO + "] " + "Message: " + message;
+                    System.out.println(logMessage);
+                } else {
+                    log(message, LoggingLevel.INFO);
+                }
+            } catch (FileMaxSizeReachedException e) {
+                displayFileMaxSizeMessages(e);
+            } catch (IOException e) {
+                System.err.println("Error occurred while logging INFO message: " + message);
+            }
+        }
+    }
+
+    public void debug(String message) {
+        if (LoggingLevel.DEBUG.ordinal() <= config.getLogLevel().ordinal()) {
+            try {
+                if (config.isLogToConsole()) {
+                    String logMessage = "[" + getTimestamp() + "][" + LoggingLevel.DEBUG + "] " + "Message: " + message;
+                    System.out.println(logMessage);
+                } else {
+                    log(message, LoggingLevel.DEBUG);
+                }
+            } catch (FileMaxSizeReachedException e) {
+                displayFileMaxSizeMessages(e);
+            } catch (IOException e) {
+                System.err.println("Error occurred while logging DEBUG message: " + message);
+            }
+        }
+    }
+
+    private void displayFileMaxSizeMessages(FileMaxSizeReachedException e) {
+        System.out.println(RED + "Maximum log file size reached or exceeded!" + RESET);
+        System.out.println(e.getLimitMessage());
+        System.out.println(e.getSizeBeforeRotationMessage());
+        System.out.println(e.getNewLogFileMessage());
+        System.out.println(e.getNewLogFilePathMessage());
+    }
+
+    public void close() {
```

Comment:
It's a bit risky to close the resource this way. 
You can make you class as FileLogger implements AutoClosable and create the FileLogger in try-with-resources block

Reply:
fixed